aten/src/ATen/core/ATenDispatch.h

```diff
@@ -0,0 +1,1261 @@
+#pragma once
+
+#include <c10/core/Backend.h>
+#include <unordered_map>
+
+// This dispatch class serves as a replacement for Type based dispatch.
```

Comment:
Can we mention here that this class is only an intermediate state and meant to be deleted when we move to c10?